### ISSUE TITLE
CMake error: CreateSourceGroup fails after loading rapidyaml with FetchContent 

### ISSUE THREAD

**Issue by kamchatka-volcano** [biojppm/rapidyaml#321]
Hello,
```
include(FetchContent)
FetchContent_Declare(ryml
    GIT_REPOSITORY https://github.com/biojppm/rapidyaml.git
    GIT_TAG ${RYML_BRANCH_NAME}
    GIT_SHALLOW FALSE  # ensure submodules are checked out
)
FetchContent_MakeAvailable(ryml)
```
on v0.4.1 this snippet taken from the example of usage of rapidyaml with FetchContent fails with the following error:

```
CMake Error at cmake-build-debug/_deps/ryml-src/ext/c4core/cmake/CreateSourceGroup.cmake:19 (string):
  string sub-command REGEX, mode REPLACE failed to compile regex
```

**Comment by biojppm** [comment]
I guess this is a duplicate of #206. 

First off, you should know that the CI tests exactly this usage of `FetchContent` scenario and it succeeds, but there are some paths which may cause it to fail. So What is the path where you have the project?

**Comment by biojppm** [comment]
FYI the fixes I mentioned are from biojppm/cmake#11, and they have been in rapidyaml since ~2months ago.

**Comment by kamchatka-volcano** [comment]
Sorry for the duplicate issue, I have exactly the same problem - "c++" folder in the build path. I used a revision tagged with v0.4.1.
